lib/functions.zsh

```diff
@@ -63,6 +63,32 @@ function try_alias_value() {
 }
 
 #
+# Alias only if not already a command/function/alias
+# Use option ALLOW_OVERRIDE_ALIASES to force overriding
+#
+# Arguments:
+#    1. my_alias='my alias value'
+#       Same argument you would pass to the "alias" built-in
+# Return value:
+#    0 if the alias was safely created, 73 otherwise
+#
+function safe_alias() {
+    local alias lhs rhs
+    alias=(${(s:=:)1})
+    lhs=$alias[1]
+    rhs=$alias[2,-1]
+    if (( $+commands[$lhs] || $+functions[$lhs] || $+aliases[$lhs] )); then
+        if [ "x$ALLOW_OVERRIDE_ALIASES" = "xtrue" ]; then
+            echo "WARN: override \"$lhs\" with alias \"$rhs\""
+	    unset ALLOW_OVERRIDE_ALIASES
```

Comment:
Wao you're faster than your own shadow! Gimme a few minutes to take your input into account ;-)